### PR TITLE
feat: adding FlameBloc mixin to allow its usage with enhanced FlameGame classes

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

Solves #1396 

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [ ] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples`.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change. (Indicate it in the [Conventional Commit] prefix with a `!`,
  e.g. `feat!:`, `fix!:`).
- [x] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database].
Indicate which of these issues are resolved or fixed by this PR, i.e. Fixes #xxxx*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

**Comment by erickzanardo** [comment]
cc @felangel for visibility 🚀

File: packages/flame_bloc/README.md [hunk 1]
```diff
@@ -15,6 +15,11 @@ BlocProvider<ExampleGame>(
 )
 ```
 
+To enable the features from `flame_bloc` into your game, you can make your game class inherit from
+`FlameBlocGame`, or if you already is using an enhanced `FlameGame` class (like fir example a
```
Comment (st-pasha) [inline packages/flame_bloc/README.md:19]:
```suggestion
`FlameBlocGame`, or if you are already using an enhanced `FlameGame` class (like for example a
```

File: packages/flame_bloc/README.md [hunk 2]
```diff
@@ -15,6 +15,11 @@ BlocProvider<ExampleGame>(
 )
 ```
 
+To enable the features from `flame_bloc` into your game, you can make your game class inherit from
```
Comment (spydon) [inline packages/flame_bloc/README.md:18]:
```suggestion
To enable the features of `flame_bloc` in your game, you can make your game class inherit from
```

**Review by spydon — APPROVED** [review]

File: packages/flame_bloc/lib/src/flame_bloc_game.dart [hunk 3]
```diff
@@ -132,3 +128,10 @@ class FlameBlocGame extends FlameGame {
     });
   }
 }
+
+/// Provides a default, concrete implementation of a [FlameBloc] game.
+class FlameBlocGame extends FlameGame with FlameBloc {
```
Comment (luanpotter) [inline packages/flame_bloc/lib/src/flame_bloc_game.dart:133]: is this still necessary? or just for backwards compat?

File: packages/flame_bloc/lib/src/flame_bloc_game.dart [hunk 4]
```diff
@@ -60,10 +60,10 @@ mixin BlocComponent<B extends BlocBase<S>, S> on Component {
   void onMount() {
     super.onMount();
     assert(
-      findGame()! is FlameBlocGame,
-      'BlocComponent can only be added to a FlameBlocGame',
+      findGame()! is FlameBloc,
+      'BlocComponent can only be added to a FlameBloc',
```
Comment (luanpotter) [inline packages/flame_bloc/lib/src/flame_bloc_game.dart:64]:
```suggestion
      'BlocComponent can only be added to a FlameBloc game',
```

**Review by luanpotter — APPROVED** [review]

File: packages/flame_bloc/lib/src/flame_bloc_game.dart (hunk 3 above)
Comment (erickzanardo) [reply packages/flame_bloc/lib/src/flame_bloc_game.dart:133]: It serves as both for backwards compatibility and as a shortcut.

**Pull request merged**